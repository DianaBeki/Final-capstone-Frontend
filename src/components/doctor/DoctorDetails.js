import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useParams, useNavigate } from 'react-router-dom';
import {
  fetchDoctorById,
  selectDoctors,
  selectStatus,
  selectError,
} from '../../redux/doctor/doctorSlice';

const DoctorDetails = () => {
  const { doctorId } = useParams();
  const dispatch = useDispatch();
  const doctors = useSelector(selectDoctors);
  const status = useSelector(selectStatus);
  const error = useSelector(selectError);
  const navigate = useNavigate();

  useEffect(() => {
    dispatch(fetchDoctorById(doctorId));
  }, [dispatch, doctorId]);

  if (status === 'loading') {
    return (
      <div>
        loading...
      </div>
    );
  }

  if (status === 'failed') {
    return (
      <div>
        <div>
          Error:
          {' '}
          {error}
        </div>
      </div>
    );
  }

  if (!doctors.length) {
    return (
      <div>
        <div>Doctor not found</div>
      </div>
    );
  }

  const doctor = doctors[0];

  return (
    <div className="flex justify-around items-center p-8">
      <img src={doctor.image} alt={doctor.name} className="w-1/2" />
      <div className="w-3/5 p-8">
        <h2 className="text-2xl font-bold">
          {doctor.name}
        </h2>
        <p>
          Specialization:
          {' '}
          <span>{doctor.specialization}</span>
        </p>
        <p>
          Fee:
          {' '}
          <span>{doctor.fee}</span>
        </p>
        <p>
          Bio:
          {' '}
          <span>{doctor.bio}</span>
        </p>
        <div>
          <button
            type="button"
            className="bg-green-500 text-white w-32 h-12 rounded-full mt-4"
            onClick={() => navigate('/reserve')}
          >
            Reserve
          </button>
        </div>
      </div>
    </div>
  );
};

export default DoctorDetails;
