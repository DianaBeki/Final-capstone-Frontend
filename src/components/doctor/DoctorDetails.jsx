import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useParams, useNavigate, Link } from 'react-router-dom';
import {
  fetchDoctorById,
  selectDoctors,
  selectStatus,
  selectError,
} from '../../redux/doctor/doctorSlice';

const DoctorDetails = () => {
  const { doctorId } = useParams();
  const dispatch = useDispatch();
  const doctors = useSelector(selectDoctors);
  const status = useSelector(selectStatus);
  const error = useSelector(selectError);
  const navigate = useNavigate();

  useEffect(() => {
    dispatch(fetchDoctorById(doctorId));
  }, [dispatch, doctorId]);

  if (status === 'loading') {
    return (
      <div>
        Loading...
      </div>
    );
  }

  if (status === 'failed') {
    return (
      <div>
        <div>
          Error:
          {' '}
          {error}
        </div>
        <div>
          Please try again later or contact support.
        </div>
      </div>
    );
  }

  if (!doctors.length) {
    return (
      <div>
        <div>
          Doctor not found
        </div>
        <div>
          <Link to="/doctors">Back to Doctors</Link>
        </div>
      </div>
    );
  }

  const selectedDoctor = doctors[0];

  return (
    <div>
      <div>
        <img src={selectedDoctor.image} alt={selectedDoctor.name} />
      </div>

      <div>
        <h2>
          {selectedDoctor.name}
        </h2>
        <p>
          Specialization:
          {' '}
          <span>{selectedDoctor.specialization}</span>
        </p>
        <p>
          Fee:
          {' '}
          <span>{selectedDoctor.fee}</span>
        </p>
        <p>
          Bio:
          {' '}
          <span>{selectedDoctor.bio}</span>
        </p>
        <div>
          <button
            className="btn btn-secondary"
            type="button"
            onClick={() => navigate('/New-reservation')}
          >
            Reserve
          </button>
        </div>
      </div>
    </div>
  );
};

export default DoctorDetails;
