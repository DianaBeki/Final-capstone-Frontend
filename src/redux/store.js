import { configureStore } from '@reduxjs/toolkit';
import doctorReducer from './doctor/doctorSlice';
import userSlice from './user/userSlice';
import reservationsReducer from './reservation/reservationSlice';

const store = configureStore({
  reducer: {
    doctor: doctorReducer,
    user: userSlice,
    reservationsList: reservationsReducer,
  },
});

export default store;
