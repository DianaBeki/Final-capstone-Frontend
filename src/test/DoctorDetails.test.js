import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { Provider } from 'react-redux';
import DoctorDetails from '../components/doctor/DoctorDetails';
import store from '../redux/store';

const render = (component) => rtlRender(
  <Provider store={store}>
    {component}
  </Provider>,
);

describe('DoctorDetails Component', () => {
  it('renders DoctorDetails component', () => {
    const { container } = render(<DoctorDetails />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
