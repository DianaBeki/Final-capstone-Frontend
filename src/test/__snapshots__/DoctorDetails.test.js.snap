// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DoctorDetails Component renders DoctorDetails component 1`] = `
<div>
  Loading...
</div>
`;
